### PR TITLE
De'Marcus Kirby Code Challenge Oct-9-2019

### PR DESCRIPTION
**Pull request by KirbyDD**

### CODE REVIEW

File: lib/house.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    @address = address
+    @rooms = []
+  end
+
+  def add_room room
+    @rooms << room
+  end
+
+  def rooms_from_category category 
+    @rooms.find_all {|room| room.category == category}
```
Comment (aziobrow) [inline lib/house.rb]: nice work choosing find_all as the right tool

File: lib/house.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def area 
+    result = 0
+    @rooms.each do |room|
+      result += room.area
```
Comment (aziobrow) [inline lib/house.rb]: looks good-- check out reduce as a cleaner alternative!

https://apidock.com/ruby/Enumerable/reduce

File: lib/house.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    end
+    result
+  end
+
+  def sort_by_area
+    result = @rooms.sort {|a, b| b.area <=> a.area}
```
Comment (aziobrow) [inline lib/house.rb]: spaceship operator! fancy! ruby has a built-in `sort_by` method, which is great here.

File: test/house_test.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
+require 'minitest'
+require 'minitest/autorun'
+require 'minitest/pride'
+require_relative '../lib/house.rb'
+require_relative '../lib/room.rb'
```
Comment (aziobrow) [inline test/house_test.rb]: require relative 😃

File: test/house_test.rb [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    @room3 = Room.new(:living_room, 25, 15)
+    @room4 = Room.new(:basement, 30, 40)
+  end
+
+  def test_it_exist
+    assert_instance_of House, @house
+  end
+
+  def test_it_has_a_price
+    assert_equal "$400000", @house.price
```
Comment (aziobrow) [inline test/house_test.rb]: Make sure to follow the interaction pattern exactly...

```pry(main)> house.price
#=> 400000
```

See how it isn't a string and doesn't have a dollar sign?

File: test/house_test.rb [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+  def test_it_has_no_rooms_by_default
+    assert_equal [], @house.rooms
+  end
+
+  def test_add_room_method
+    @house.add_room(@room)
+    assert_equal [@room], @house.rooms
+    @house.add_room(@room2)
+    assert_equal [@room, @room2], @house.rooms
```
Comment (aziobrow) [inline test/house_test.rb]: love the progression here, nice work

File: test/house_test.rb [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+  def test_rooms_from_category_method
+    @house.add_room(@room)
+    @house.add_room(@room2)
+    @house.add_room(@room3)
+    @house.add_room(@room4)
+
+    assert_equal [@room, @room2], @house.rooms_from_category(:bedroom)
+    assert_equal [@room3], @house.rooms_from_category(:living_room)
+    assert_equal [@room4], @house.rooms_from_category(:basement)
```
Comment (aziobrow) [inline test/house_test.rb]: nice job testing multiple categories. What happens if they pass in a category that doesn't exist? consider testing those edge cases when you can.

File: test/house_test.rb [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  def test_area_test
+    @house.add_room(@room)
+    @house.add_room(@room2)
+    assert_equal 290, @house.area
+
+    @house.add_room(@room3)
+    assert_equal 665, @house.area
+
+    @house.add_room(@room4)
+    assert_equal 1865, @house.area
```
Comment (aziobrow) [inline test/house_test.rb]: again, great job testing the progression!

File: test/house_test.rb [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+  end
+
+  def test_rooms_by_category_method
+    skip
+    @house.add_room(@room)
+    @house.add_room(@room2)
+    @house.add_room(@room3)
+    @house.add_room(@room4)
+    expectedResult = { :bedroom => [@room1, @room2], :living_room => [@room3], :basement => [@room4]}
+    assert_equal expectedResult, @house.rooms_by_category
```
Comment (aziobrow) [inline test/house_test.rb]: i like your TDD here! This was a tricky one.
